### PR TITLE
Added options for discord avatars

### PR DESCRIPTION
**Pull request by stubbo**
Thought it would be a good idea to allow some choice for the avatar being received as discord offer a large range.

[Discord Docs](https://discord.com/developers/docs/reference#image-formatting-cdn-endpoints)

Not sure if I am doing this contribution correctly but didnt see anything on the docs about doing updates 😛

### CODE REVIEW

**Comment by TheRealHeyJack** [comment]
good one my g

File: src/Discord/Provider.php [hunk 1]
```diff
@@ -61,6 +61,24 @@ protected function getUserByToken($token)
...
     }
 
+    /**
+     * @param array $user
+     *
+     * @return string|null
+     */
+    protected function formatAvatar(array $user)
+    {
+        if (is_null($user['avatar'])) {
```
Comment (atymic) [inline src/Discord/Provider.php:71]:
```suggestion
        if (empty($user['avatar'])) {
```

File: src/Discord/Provider.php [hunk 2]
```diff
@@ -61,6 +61,24 @@ protected function getUserByToken($token)
...
+            return null;
+        }
+
+        $isGif = preg_match('/a_.+/m', $user['avatar']) === 1;
```
Comment (atymic) [inline src/Discord/Provider.php:75]: Can you explain this please?

File: src/Discord/Provider.php [hunk 3]
```diff
@@ -61,6 +61,24 @@ protected function getUserByToken($token)
...
+        $extension = $this->getConfig('avatar_gifs', true) && $isGif ? 'gif' :
```
Comment (atymic) [inline src/Discord/Provider.php:76]:
```suggestion
        $extension = $this->getConfig('allow_gif_avatars', true) && $isGif ? 'gif' :
```

File: src/Discord/Provider.php [hunk 4]
```diff
@@ -61,6 +61,24 @@ protected function getUserByToken($token)
...
+            $this->getConfig('avatar_default_extension', 'webp');
```
Comment (atymic) [inline src/Discord/Provider.php:77]: Default as `jpg` to not break BC please

File: src/Discord/Provider.php [hunk 5]
```diff
@@ -61,6 +61,24 @@ protected function getUserByToken($token)
...
+
+        return sprintf('https://cdn.discordapp.com/avatars/%s/%s.%s', $user['id'], $user['avatar'], $extension);
```
Comment (atymic) [inline src/Discord/Provider.php:79]: Will this always work? Is it possible our dynamic URL could 404?

**Review by atymic — CHANGES_REQUESTED** [review]

File: src/Discord/Provider.php (hunk 2 above)
Comment (stubbo) [reply src/Discord/Provider.php:75]: As explained on the docs for discord animated/gif icon starts with a_

File: src/Discord/Provider.php (hunk 5 above)
Comment (stubbo) [reply src/Discord/Provider.php:79]: As long as the default is either jpg, png or webp there shouldnt be any 404's.

Comment (stubbo) [reply src/Discord/Provider.php:79]: I can explain a little bit more in the readme if wanted

File: src/Discord/Provider.php (hunk 1 above)
Comment (stubbo) [reply src/Discord/Provider.php:71]: I will get this changed, just carried on with what was already in the file

File: src/Discord/Provider.php (hunk 5 above)
Comment (atymic) [reply src/Discord/Provider.php:79]: No worries, `jpg` please

Could you add an `@see` docblock tag with a link to the discord docs?

File: src/Discord/Provider.php (hunk 2 above)
Comment (atymic) [reply src/Discord/Provider.php:75]: Thanks, see comment below :)

File: src/Discord/Provider.php (hunk 1 above)
Comment (atymic) [reply src/Discord/Provider.php:71]: thanks :)

File: src/Discord/Provider.php (hunk 5 above)
Comment (atymic) [reply src/Discord/Provider.php:79]: Would be good to document in the readme a bit more as well, thanks!

Comment (stubbo) [reply src/Discord/Provider.php:79]: > Could you add an `@see` docblock tag with a link to the discord docs?
Will add that now as well

**Review by atymic — APPROVED** [review]

**Pull request merged**